blog/content/post/2020-04-30-ipxe-https.md

```diff
@@ -0,0 +1,216 @@
+
+iPXE (http://ipxe.org) 01:00.0 C000 PCI2.10 PnP PMM+CFE3C6F0+CFD7C6F0 C000
+```
+
+When the iPXE prompt appears, press `Ctrl+B` to enter iPXE shell:
+
+```
+iPXE (PCI 01:00.0) starting execution...ok
+iPXE initialising devices...ok
+
+
+
+iPXE 1.0.0+ (ebf2) -- Open Source Network Boot Firmware -- http://ipxe.org
+Features: DNS HTTP HTTPS iSCSI TFTP AoE ELF MBOOT PXE bzImage Menu PXEXT
+
+iPXE>
+```
+
+Now you have variety of options:
+
+- Boot network installation of your favorite system,
+- Boot kernel and initrd directly to diskless systems using NFS,
+- etc.
+
+Example menu entries used by 3mdeb are available on [3mdeb GitHub](https://github.com/3mdeb/netboot/blob/master/menu.ipxe)
+Those are HTTP only. To boot such menu one has to setup the [3mdeb PXE server](https://github.com/3mdeb/pxe-server)
+which uses the netboot repository and invoke the following from iPXE shell:
+
+```bash
+iPXE> dhcp net0
+Configuring (net0 00:0d:b9:51:fc:1c)...... ok
+iPXE> chain http://<pxe-server-ip>:<port>/menu.ipxe
+```
+
+![iPXE menu](/img/pxe_menu.png)
+
+One can invoke Debian network installation or direct booting to diskless
+systems.
+
+> You may also use autoboot command which will boot form the provided DHCP
+> options, but this requires proper DHCP configuration providing the IP
+> address, and path to the desired image to be loaded and executed.
+
+## Booting with HTTPS
+
+As you may noticed in previous section, it is possible to boot using HTTPS (see
+features):
+
+```bash
+Features: DNS HTTP HTTPS iSCSI TFTP AoE ELF MBOOT PXE bzImage Menu PXEXT
+```
+
+3mdeb has a HTTPS server which hosts a simple menu for development of various
+projects and platforms (as it is only for development purposes the options in
+menu presented below may not necessarily work each time).
+
+To boot 3mdeb menu invoke the commands in iPXE shell:
+
+```bash
+iPXE> chain https://boot.3mdeb.com/menu.ipxe
+https://boot.3mdeb.com/menu.ipxe.... ok
+```
+
+Sometimes during the menu.ipxe file loading, iPXE shortly prints the
+certificate name from Let's Encrypt.
+
+![3mdeb iPXE menu](/img/3mdeb_pxe_menu.png)
+
+The above proves that HTTPS works on iPXE. However, there are still issues with
+certain certificates. In the default configuration, iPXE trusts only a single
+root certificate: the `iPXE root CA` certificate. This root certificate is used
+to cross-sign the standard Mozilla list of public CA certificates. In the
+default configuration, iPXE will therefore automatically trust the same set of
+certificates as the Firefox web browser. More details on [iPXE documentation](https://ipxe.org/crypto)
+
+## Future
+
+In the nearest future I will show you how to utilize the `imgverify` iPXE
+command used to verify downloaded images. This article is just a beginning. Our
+plan is to support booting TrenchBoot enabled images over HTTPS with image
+verification. In the next posts I will show you how to verify images and embed
+own iPXE scripts so stay tuned.
```

Comment:
Broken syntax in the paragraph. Try sth like:
This article is just a beginning. Our plan is to support booting TrenchBoot enabled images over HTTPS with image verification. In the series I will show you how to utilize the `imgverify` iPXE command used to verify downloaded images and how to verify images and embed your own iPXE scripts,  so stay tuned.

Reply:
Btw, I like your style, it's really nice article!